packages/next/src/build/index.ts

```diff
@@ -368,6 +368,42 @@ function getCacheDir(distDir: string): string {
   return cacheDir
 }
 
+function getNumberOfWorkers(config: NextConfigComplete) {
+  if (
+    config.experimental.cpus &&
+    config.experimental.cpus !== defaultConfig.experimental!.cpus
+  ) {
+    return config.experimental.cpus
+  }
+
+  if (config.experimental.memoryBasedWorkersCount) {
+    return Math.max(
+      Math.min(config.experimental.cpus || 1, Math.floor(os.freemem() / 1e9)),
+      // enforce a minimum of 4 workers
+      4
+    )
+  }
+
+  if (config.experimental.cpus) {
+    return config.experimental.cpus
+  }
+
+  // Fall back to 4 workers if a count is not specified
+  return 4
+}
+
+async function writeFileUtf8(filePath: string, content: string) {
```

Comment:
is there a particular point to this?

Reply:
Pretty straightforward, not all writefile calls had utf-8 set even though they should have, this makes sure all of them have it.